src/cpp/parse.cpp

```diff
@@ -479,9 +479,9 @@ void Parser::merge_types(const typet &src, typet &dest)
     // the end of the subtypes container needs to stay the same,
     // since several analysis functions traverse via the end for
     // merged_types
-    typet::subtypest &sub=dest.subtypes();
+    auto &sub = to_type_with_subtypes(dest).subtypes();
     sub.emplace(sub.begin(), src);
-    POSTCONDITION(!dest.subtypes().empty());
+    POSTCONDITION(!to_type_with_subtypes(dest).subtypes().empty());
```

Comment:
What reason do we have to believe that this would ever be violated? It seems we are checking that `emplace` does what it claims?